### PR TITLE
Initial documentation on session configuration

### PR DESCRIPTION
**Pull request by GawainLynch**
This is option B) for https://github.com/bolt/bolt/pull/6771 and heavily documents available storage configuration, give examples, links to upstream as appropriate, and warnings & advice notes where appropriate/important.

I've initially aimed this at 3.2 as I seem to remember there being problems earlier, if I am wrong, let me know and I will target either 3.0 or 3.3.

Also this is basically written from the top of my head, with looks at 3.2 & 3.3 SPs, but not tested … so :koala:

### CODE REVIEW

File: docs/configuration/advanced/sessions.md [hunk 1]
```diff
@@ -0,0 +1,392 @@
+---
+title: Sessions
+level: advanced
+---
+Sessions
+========
+
+By default, Bolt's sessions are managed behind the scenes for you by an
```
Comment (bobdenotter) [inline docs/configuration/advanced/sessions.md]: "A"

File: docs/configuration/advanced/sessions.md [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+configurable abstraction layer.
+
+Configuration options are selected based on an order of precedence from varying
+locations, and merges options together.
```
Comment (bobdenotter) [inline docs/configuration/advanced/sessions.md]: This looks odd at first glance. Who or what does this merging?

File: docs/configuration/advanced/sessions.md [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+These defaults are limited to those that are useful, and secure.
+
+
+<p class="note"><strong>Note:</strong> Under most circumstances you should not
+need to adjust these settings.</p>
+
+
+## Browser cookies
+
+By default, Bolt wil inherit the settings `cookies_lifetime:`, `cookies_domain`,
```
Comment (bobdenotter) [inline docs/configuration/advanced/sessions.md]: No semicolon here in cookies_lifetime

File: docs/configuration/advanced/sessions.md [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+and `enforce_ssl` (for `cookie_secure`) should no overide options be set, as
+per the order of precedence explained in the introduction.
+
+However, there are several override settings avalailable, should you need more
```
Comment (bobdenotter) [inline docs/configuration/advanced/sessions.md]: Available

File: docs/configuration/advanced/sessions.md [hunk 5]
```diff
@@ -0,0 +1,392 @@
...
+Should you encounter exceptions from <code>SessionServiceProvider</code>
+indicating problems with PHP's system save path, set
+<code>save_handler: filesystem</code>, and the <code>save_path</code> option
+shown below.</p>
+
+
+## Saved session file path
+
+Session data is cached in between requests. Bolt puts these files in a hidden
+directory of the cache, and **is not** cleared by byt the normal cache clearing
```
Comment (bobdenotter) [inline docs/configuration/advanced/sessions.md]: "By"

File: docs/configuration/advanced/sessions.md (hunk 2 above)
Comment (GawainLynch) [inline docs/configuration/advanced/sessions.md]: > By default, Bolt's sessions are managed behind the scenes

Comment (GawainLynch) [inline docs/configuration/advanced/sessions.md]: If you haven't groked that at a reader, this page is not intended for you.

**Comment by GawainLynch** [comment]
> I don't think we need to give an example of yaml config and DI injection for each value. Just give it once and then describe what each key is for.

Yeah, good call … much more consistent now with the rest of the docs.

> I think the most important aspect of this configuration is the different handlers and how they are configured, which has yet to be covered.

OK, so spent the last 5 hours going over our code, Symfony's, and documentation of Redis & Memcached, and hopefully added what you're more-or-less after … enough scope creep versus bolt/bolt#6771 on a feature that landed in #3900 nearly two years ago, I think :wink:

**Comment by GawainLynch** [comment]
Oh, and given that I got no answers, I examined our SPs back to 3.0 and the changes between then and 3.2 are negligible, so I've moved this down to `release/3.0`.

**Comment by GawainLynch** [comment]
B.:princess:.F.L-ing

**Pull request merged**